utils/global_events.cfg

```diff
@@ -1992,3 +1992,111 @@ $soul_eating"
         [/if]
     [/event]
 #enddef
+
+#define POISON_SWAMP SPEAKER1 SPEAKER2
+    [event]
+        name=new turn
+        id=poison_swamp_1st_warning
+        [filter_condition]
+            [have_unit]
+                side=1
+                [filter_location]
+                    terrain=Ss,Ss^*,Chs,Khs,*^Vhs
+                [/filter_location]
+            [/have_unit]
+        [/filter_condition]
+        [message]
+            speaker={SPEAKER1}
+            message=_"The swamp causes disease. Our soldiers should avoid standing in it for longer than absolutely necessary."
+        [/message]
+        [message]
+            speaker=Delly
+            message=_"This is why I <span font='italic'>hate</span> swamps!"
+            [show_if]
+                [have_unit]
+                    id=Delly
+                [/have_unit]
+            [/show_if]
+        [/message]
+        [event]
+            name=new turn
+            id=poison_swamp_2nd_warning
+            [filter_condition]
+                [have_unit]
+                    side=1
+                    [filter_location]
+                        terrain=Ss,Ss^*,Chs,Khs,*^Vhs
+                    [/filter_location]
+                    [filter_wml]
+                        [status]
+                            poisoned=yes  # Note: you won't get second warning until a unit that is poisonable spends a second turn in swamp.
+                        [/status]
```

Comment:
Can this cause performance slowdowns on new turn in practice? `[filter_wml]` is claimed to be quite slow unless used on the unit's variables.

Reply:
I don't know.  I would think that the first two filters would have to match (side 1 unit in swamp) before the third would be evaluated, and then hopefully only for units that matched the first conditions.  And I read that filter_wml does not expand units unless necessary (is it for status?).  I just copied most of this from AE's Dead Menance.  I certainly didn't notice any slowdown in the scenarios where it was used.

Reply:
If you didn't observe anything, it's probably fine.